src/lwpa/common.c

```diff
@@ -20,55 +20,71 @@
 #include "lwpa/common.h"
 
 #include "lwpa/bool.h"
-#include "lwpa/private/common.h"
+
 #include "lwpa/private/log.h"
 #include "lwpa/private/netint.h"
+#include "lwpa/private/socket.h"
+#include "lwpa/private/timer.h"
 
-lwpa_error_t lwpa_init(lwpa_features_t features)
+typedef struct LwpaModuleInit
 {
-  bool os_initted = false;
-  bool logging_initted = false;
+  bool initted;
+  lwpa_error_t (*init_fn)();
+  void (*deinit_fn)();
+} LwpaModuleInit;
 
-  // Step 1: OS-specific init
-  lwpa_error_t res = lwpa_os_init(features);
-  os_initted = (res == kLwpaErrOk);
+#define LWPA_MODULE_INIT(module_name) { false, module_name##_init, module_name##_deinit }
 
-  // Step 2: OS-neutral init
-  if (res == kLwpaErrOk)
+#define LWPA_MODULE_INIT_ARRAY \
```

Comment:
There be some fun macro voodoo in here & the previous #define.  It took me a minute to figure out what you were doing. :)

Reply:
I'll look at making it more obvious either with comments or nicer code.

Reply:
Now that I understand, I think the macros are fine. Maybe a quick comment.